### ISSUE TITLE
[BUG] Kill command does not kill sub-subprocess groups

### ISSUE THREAD

**Issue by davidlin409** [Nukesor/pueue#347]
OS: Ubuntu 20.0.4

Pueue's design is really great, and recently I tries to run experiment using pueue. My experiment is to run bash script, which calls 3 different other bash scripts. Under-lying script will then call Python to run actually experiment.

When I tries to kill the experiemnt, bash script is killed, but the Python process, which is the actual program to run, is not killed. 

On the other hand, when I am using task spooler, kill process can really kill all process, including Python instance, and process run by Python.

**Comment by Nukesor** [comment]
All `pueue` does, is to send a `SIGTERM` to the subprocess.

The subprocess itself should then go ahead and propagate the unix signal.
However, since quite a few processes don't do this properly, `pueue kill` has a `-c` flag, which not only sends the kill to the main subprocess, but also to subprocesses of the subprocess.

This shouldn't e the default behavior though, as this isn't how Unix signals are supposed to work.

tldr:
Just use the `--children/-c` flag

**Comment by Nukesor** [comment]
Also note, that this behavior doesn't work, if you're on MacOs, as there're no good process handling libraries for MacOs.

I only write this in case somebody with a Mac stumbles upon the same problem ;)

**Comment by Nukesor** [comment]
I'll mark this as resolved for now :)

If you have any further questions, just comment on this Issue!
